qemu/tests/chardev_free_port.py

```diff
@@ -0,0 +1,46 @@
+from virttest import error_context, env_process
+from virttest import utils_misc
+
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    qemu should try to find a free port value by to=<port> with unix socket and tcp options:
+    1) boot guest with socket with -qmp tcp:0:4444.
+    2) telnet localhost.
+    3) check output of {"execute": "qmp_capabilities"}, {"execute":"query-chardev"}.
+    4) repeat step 1 to 3 with -qmp tcp:0:4445 and telnet to port 4445.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    chardev_infos = []
+    port_nums = []
+    vms = params.get("vms").split()
+    _vm = env.get_vm(vms[0])
+    char_devices = add_chardev(_vm, params)
+    chardev_params = char_devices[0].params
+    for vm_ind, vm in enumerate(vms):
+        if vm_ind == 1:
+            host = chardev_params['host']
+            chardev_to = utils_misc.find_free_ports(int(port_nums[0]) + 1, 6000, 1, host)
+            chardev_params['to'] = str(chardev_to[0])
+        for device in char_devices:
+            extra_params = ' ' + device.cmdline()
+            params['extra_params_%s' % vm] = params.get('extra_params', '') + extra_params
+        params['start_vm_%s' % vm] = "yes"
+        env_process.preprocess(test, params, env)
+        _vm = env.get_vm(vm)
+        chardev_infos.append(_vm.monitor.info("chardev"))
+        port_nums.append(str(int(chardev_params['port'])+vm_ind))
+
+    for char_ind, chardevs in enumerate(chardev_infos):
+        in_chardev = False
+        for chardev in chardevs:
+            if port_nums[char_ind] in chardev['filename']:
```

Comment:
together with L38, @nanliu-r if boot the second vm with 'port=a, to=b' (b > a here), the actual port number for the second vm always be `a+1` ?

Reply:
I think it is no, this is not the check point for this case,  the second vm can boot up normally is ok IMO.

Reply:
It isn't, the checkpoint is we can find a free port in the range `[a, b]`, and we do not limit the value of `b`, but I think it's better to set `b` larger than `a`. because it is close to the test environments, what do you think?

Reply:
As we discussed, the actual port of the second vm is between the "port" and "to", it is not must to be `"port" + 1`, @michiboo  please update, thanks!